app/assets/javascripts/discourse/app/components/composer-editor.js

```diff
@@ -630,6 +630,51 @@ export default Component.extend(ComposerUpload, {
     });
   },
 
+  _registerImageAltTextButtonClick($preview) {
+    $preview
```

Comment:
I'm not sure how much time yet you have to allocate on this, but it would be a good exercice to remove jquery usage of this function. We should do it with scaling too but that's outside the scope of this PR.

It should be rather simpler to do. 

One harder thing to do would be to think about the listeners here, I'm not a fan of this off/on pattern, maybe try to see if you can think of a place to have a better hook to cleanly removeEventListener. No worries if you can't find a way, I know this is how we currently do in core.

Reply:
I think there are quite a few techniques we can employ but they might seem quite contrived. Perhaps something like [one](https://api.jquery.com/one/) and also maybe we can make use of event namespaces (once again, jQuery specific). More investigation needs to be done for sure.